v2.7

anjakefala opened the pull request.

saulpw committed:
[plugins] fail silently if error retrieving plugins list

saulpw committed:
[dev] cleanup plugins imports

saulpw committed:
[vdplus] remove automatic import

anjakefala committed:
[dev] bump to develop version

anjakefala committed:
[dev] Update debian to 2.6.1

anjakefala committed:
[plugins] fix hash for dedupe

anjakefala committed:
[slide-] fix Shift slide-down and Shift slide-up with arrow keys

Closes #1137

anjakefala committed:
[docs] remove out-dated pages

saulpw committed:
[api clipboard] add memory.clipcols for columns

- saved by copyRow so plugins can get table of clipboard values using memory.cliprows+memory.clipcols

saulpw committed:
[AttrDict-] fix bug with nested AttrDict

if a value is explicitly set to AttrDict, then the result is not
    wrapped (making a copy), so values can be set on sub-dicts
    without disappearing

saulpw committed:
[draw] fixed-height multiline rows #916

saulpw committed:
[api] addCommand return longname for chaining

anjakefala committed:
[replay threads] do not block progress for replay thread

- commands like `join`, wait for other threads, e.g. loading threads, to complete, before
progressing
- replay progression has its own thread in live-mode. we do not want
replay thread to block the progression of commands that it is execution.
- if `noblock` is set on a thread, `sync()` will not wait for that
thread to complete before continuing
- so replay() will not block other threads

Closes #1141

saulpw committed:
[options cleanup] default_height/width per sheet

saulpw committed:
[menu] standardize lowercase text on 3rd+ level of menu

saulpw committed:
[macos-] fix bindings for option+key

anjakefala committed:
[docs] note how to use vd as a pager within psql (thanks @radostyle)

Closes #1113

anjakefala committed:
[docs] note how to load a hierarchy of parquet files

Closes #1126

anjakefala committed:
Revert "[__init__] for the moment, revert to importing globals"

This reverts commit 651ec752c82a70f7538f57d0c8f6d75e1dc1c838.

We are now removing support for importing globals

saulpw committed:
[menu] display as much of disp_menu_fmt as possible

saulpw committed:
[json api] add vd.encode_json(row, cols)

anjakefala committed:
[start-up] do not raise warning, if plugins dir does not exist

Closes #1131

anjakefala committed:
[dev] have dev/mkman.sh build docs/man.md

Closes #1130

saulpw committed:
Merge pull request #1152 from saulpw/plugins-warning

[start-up] create plugins dir, if does not exist

anjakefala committed:
[help-pane-] move input help pane from vdplus to visidata

anjakefala committed:
[default_sample_size] provide warning when sample rows != number of rows in the sheet

Closes #1054

davidskeck committed:
Updated sqlite handler to wrap column names in quotes (#1157)

Any column names which included spaces would cause the following error on line 208: sqlite3.OperationalError: near "test": syntax error
This fix wraps the column names in quotes, which is what the sqlite documentation suggests to do when columns have spaces.

saulpw committed:
[status] do not keep history of debug messages if debug=False

anjakefala committed:
[keys-] move longer keys above shorter keys

for k, v in vd.prettykeys_trdict.items() will result in a conflict for
    partially overlapping keys (e.g. KEY_SR and KEY_SRIGHT

anjakefala committed:
[sqlite-] allow encoding/decoding of sqlite db with invalid unicode

Closes #1156

anjakefala committed:
[tests] add test for #1156 (thanks @WesleyAC)

anjakefala committed:
Merge pull request #1158 from saulpw/sqlite_encoding

[sqlite-] allow encoding/decoding of sqlite db with invalid unicode

saulpw committed:
[sheets-] fix error for colorizing selected rows in some cases

saulpw committed:
[async] allow cancel of async methods on Sheet #1136

saulpw committed:
[async-] fix batch mode execAsync to include sheet= kwarg #1136

saulpw committed:
[random-] import into globals for some commands #1162

saulpw committed:
[movement] bind Home/End to go-top/bottom #1161

anjakefala committed:
[test] add xlsx sample data with merged and coloured cells

anjakefala committed:
[test] add sample data for EmptyCell, colored cells, and merged cells

Commit:
[xlsx-] add handling of EmptyCell instances

See 53c9d7c: [xlsx] change rowdef to list of AttrDict

Closes #1121

anjakefala committed:
[test] add test for 1121

saulpw committed:
[locale] warn instead of abort for invalid locale

saulpw committed:
[sqlite-] use options.encoding_errors to decode text #1156

saulpw committed:
[options] disallow editing option name column #1132

saulpw committed:
[xlsx-] options.clean_names for sheet name at save #594 #1122

saulpw committed:
[sheet api] Sheet.names for name components #1122

will be useful for roundtripping multisheet formats like xlsx

anjakefala committed:
[docs] update recommendation from options.expand_col_scanrows to options.default_sample_size

Commit:
Move urlcache to vd object (#1164)

* moved urlcache to vd object
* Added as global_api

saulpw committed:
[expand-cols] warn when sampling fewer than all rows #1054

saulpw committed:
[splitwin] minimum of 3 lines for primary window #1166

hoclun-rigsep committed:
[xlsx] Stop the loader choking on EmptyCell instances

See 53c9d7cd: [xlsx] change rowdef to list of AttrDict

I believe this chokes when it encounters a row with a MergedCell:
the 'column_letter' lookup fails on instances of EmptyCell, which follow
the MergedCell.

The present commit is one way to address that.

hoclun-rigsep committed:
[xlsx] Offer some optional columns

Adds the following options:

    vd.option('xlsx_cellobject', False, 'include column of cell objects', replay=True)
    vd.option('xlsx_fontcolor', False, 'include cell font colors', replay=True)
    vd.option('xlsx_fillcolor', False, 'include cell fill colors', replay=True)

saulpw committed:
[ods] add loader for odf spreadsheets

anjakefala committed:
[unfurl-col-] fix unfurl-col with exceptions (thanks @jsvine)

Closes #1171

anjakefala committed:
[tests] add tests for #1171

markus1189 committed:
Fix typo in description of open-colors help

anjakefala committed:
[FileProgress-] fix corruption when saving to compression formats # 1159

This revert is a band aid. and probably breaks Progress counters for compression
formats. #1175 was opened to investigate a better fix.

anjakefala committed:
[FileProgress] add clarifying comments

pstuifzand committed:
[dup-*-deep-] fix error with async_deepcopy

anjakefala committed:
Merge pull request #1173 from markus1189/typo

Fix typo in description of open-colors help

saulpw committed:
[commands-] exec-command should work on all sheets

saulpw committed:
[draw] clarify notewidth is 1

anjakefala committed:
Merge pull request #1098 from hoclun-rigsep/develop

Improved XLSX loader

anjakefala committed:
[xlsx] simplify xlsx meta columns to single option xlsx_meta_columns

Affects #1098

anjakefala committed:
[tests] add test for #1098 and #1121

anjakefala committed:
[test] fix test

anjakefala committed:
[status] add space to the right of keystrokes

Closes #1176

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/saulpw/visidata?pullRequest=1178) <br/>All committers have signed the CLA.

anjakefala committed:
[dev] update CHANGELOG

anjakefala committed:
[test] add test for loading .ods

anjakefala committed:
[tests] clean up xlsx special cases tests

anjakefala committed:
[cmdlog-] do not log cmdlog-sheet in global CommandLog

saulpw committed:
[awk] new loader for simple awk-like records #1179

saulpw committed:
[open-row] show column headers #1172

Commit:
Check SHA256 if provided

anjakefala committed:
[cache] clear caches before saving

Caches are cleared between commands (in execCommand), However, in a batch-output case (`-b
-o`), no commands are issued. For this case, important to clear before
saving.

Closes #1181

saulpw committed:
[save-selected] better default sheet name

anjakefala committed:
Merge pull request #1183 from geekscrapy/maycheck256

Check SHA256, if provided

saulpw committed:
[loaders] rename awk filetype to lsv #1179

anjakefala committed:
[plugins] update to newer version of darkdraw

anjakefala committed:
[save-] vd.fail when saving multiple sheets in a single non-embeddable format

- a directory is required

Bug reported in #1180

saulpw committed:
[undo-] call undofuncs in reverse order #1133

Both setValues and setValue call addUndo, and setValues calls setValue.
So multiple undofuncs are added for some commands, like delete-cell.
Calling these undofuncs as implemented and in the order they are added, causes some internal confusion leading to this warning message, but actually nothing is incorrect and the results are the same.
Ideally, commands would not add more undofuncs than necessary (it can waste memory), but that's trickier than this fix, which just calls undofuncs in reverse LIFO order, so that subsequent undofuncs undo previous themselves without concern.

saulpw committed:
[windows prettykeys] map ^M to Enter #1154

Possible regression of #741 due to prettykeys.

saulpw committed:
[join-] use getDisplayValue for joinkey value, like groupRowsByKey #1124

anjakefala committed:
[tests] add test for #1124

anjakefala committed:
[test] add golden

Commit:
Do not load if sheet.load_lazy is True or global load_lazy is True

Commit:
dont continue if main package fails pip install

anjakefala committed:
Merge pull request #1194 from geekscrapy/no_install_if_pip_failure

dont continue if main package fails pip install

anjakefala committed:
Merge pull request #1193 from geekscrapy/sheet_load_lazy

Do not auto load if sheet.load_lazy is True or global load_lazy is True

anjakefala committed:
[save] confirm when saving to options.save_filetype [default now 'tsv']

Closes #1180

saulpw committed:
[sqlite] query/insert WITHOUT ROWID tables; no modify/delete #1111

saulpw committed:
[help-] move isLoggableCommand to vd obj for commands help sheet

anjakefala committed:
[status-] fix colouring for inactive status pane

- its precedence was not high enough to trump color_status

anjakefala committed:
[tests] add test for #1111

anjakefala committed:
Merge branch 'stable' into develop

anjakefala committed:
[awk] rename awk to .lsv #1179

anjakefala committed:
[dev] update CHANGELOG

anjakefala committed:
[formats] add section for .lsv

saulpw committed:
[dev] update CHANGELOG

anjakefala committed:
[pandas-] adapt 9e4aa7c9b806 for pandas formats

Truth value for a Series object is ambiguous.

anjakefala committed:
[pandas-] fix frequency table in pandas regression since 444299261

anjakefala committed:
[dev] bump to v2.7; update manpage
